arangod/RocksDBEngine/RocksDBMetadata.cpp

```diff
@@ -292,13 +293,19 @@ Result RocksDBMetadata::serializeMeta(rocksdb::WriteBatch& batch,
   TRI_ASSERT(!coll.isAStub());
   TRI_ASSERT(appliedSeq != UINT64_MAX);
   TRI_ASSERT(appliedSeq > 0);
+    
+  TRI_ASSERT(batch.Count() == 0);
 
   Result res;
   if (coll.deleted()) {
     return res;
   }
+    
+  auto& engine = coll.vocbase().server().getFeature<EngineSelectorFeature>().engine<RocksDBEngine>();
+  std::string const context = coll.vocbase().name() + "/" + coll.name();
 
-  const rocksdb::SequenceNumber maxCommitSeq = committableSeq(appliedSeq);
+  rocksdb::SequenceNumber const maxCommitSeq = committableSeq(appliedSeq);
+  TRI_ASSERT(maxCommitSeq <= appliedSeq);
```

Comment:
I think this is unnecessarily strict. The `appliedSeq` we get passed in here is the min seq number of all collections processed so far, but IMO it doesn't make sense to limit our own comittableSeq by the seq nr from other collections. If this is indeed necessary, then this is very fishy since we depend on the order in which the collections are processed.